pkg/xray/resource_xray_repository_config.go

```diff
@@ -0,0 +1,349 @@
+										Description:      `The artifact will be retained for the number of days you set here, after the artifact is scanned. This will apply to all artifacts in the repository.`,
+										ValidateDiagFunc: validator.IntAtLeast(0),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	)
+
+	var unpackPattern = func(s []interface{}) []Pattern {
+		var patterns []Pattern
+
+		for _, raw := range s {
+			data := raw.(map[string]interface{})
+			pattern := Pattern{
+				Include:           data["include"].(string),
+				Exclude:           data["exclude"].(string),
+				IndexNewArtifacts: data["index_new_artifacts"].(bool),
+				RetentionInDays:   data["retention_in_days"].(int),
+			}
+			patterns = append(patterns, pattern)
+		}
+
+		return patterns
+	}
+
+	var unpackAllOtherArtifacts = func(config *schema.Set) AllOtherArtifacts {
+		allOtherArtifacts := AllOtherArtifacts{}
+
+		if config != nil {
+			for _, raw := range config.List() {
+				data := raw.(map[string]interface{})
+				allOtherArtifacts.IndexNewArtifacts = data["index_new_artifacts"].(bool)
+				allOtherArtifacts.RetentionInDays = data["retention_in_days"].(int)
+			}
+		}
+
+		return allOtherArtifacts
+	}
+
+	var unpackRepoPathConfig = func(config *schema.Set) *RepoPathsConfiguration {
+		repoPathsConfiguration := new(RepoPathsConfiguration)
+		configList := config.List()
+		if len(configList) == 0 {
+			return nil
+		}
+
+		m := configList[0].(map[string]interface{})
+
+		otherArtifacts := unpackAllOtherArtifacts(m["all_other_artifacts"].(*schema.Set))
+		repoPathsConfiguration.OtherArtifacts = otherArtifacts
+
+		patterns := unpackPattern(m["pattern"].([]interface{}))
+		repoPathsConfiguration.PatternList = patterns
+
+		return repoPathsConfiguration
+	}
+
+	var unpackRepoConfig = func(config *schema.Set) *GeneralRepoConfig {
+		repoConfig := new(GeneralRepoConfig)
+
+		if config != nil {
+			for _, raw := range config.List() {
+				data := raw.(map[string]interface{})
+				repoConfig.VulnContextualAnalysis = data["vuln_contextual_analysis"].(bool)
+				repoConfig.RetentionInDays = data["retention_in_days"].(int)
+			}
+		}
+
+		return repoConfig
+	}
+
+	var unpackRepositoryConfig = func(s *schema.ResourceData) RepositoryConfiguration {
+		d := &util.ResourceData{ResourceData: s}
+
+		repositoryConfig := RepositoryConfiguration{
+			RepoName: d.GetString("repo_name", false),
+		}
+
+		if _, ok := s.GetOk("repo_config"); ok {
+			repoConfig := unpackRepoConfig(s.Get("repo_config").(*schema.Set))
+			repositoryConfig.RepoConfig = repoConfig
+		}
+
+		if _, ok := s.GetOk("repo_paths_config"); ok {
+			repoPathConfig := unpackRepoPathConfig(s.Get("repo_paths_config").(*schema.Set))
+			repositoryConfig.RepoPathsConfig = repoPathConfig
+		}
+		return repositoryConfig
+	}
+
+	var packGeneralRepoConfig = func(repoConfig *GeneralRepoConfig) []interface{} {
+		if repoConfig == nil {
+			return []interface{}{}
+		}
+
+		m := map[string]interface{}{}
```

Comment:
Let's keep style consistent and use inline initialization here.

Reply:
Replaced with inline initialization.